src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -162,22 +191,44 @@ public void RunScenario(ScenarioConfiguration configuration, Func<ScenarioBenchm
                     }
                     else
                     {
-                        scenarioExecutionResult = Run(configuration, scenario);
+                        scenarioExecutionResult = Run(configuration, scenarioTest);
                     }
 
                     configuration.PostIterationDelegate?.Invoke(scenarioExecutionResult);
                 }
             }
 
-            ScenarioBenchmark scenarioBenchmark = teardownDelegate();
-            if (scenarioBenchmark == null)
-                throw new InvalidOperationException("The Teardown Delegate should return a valid instance of ScenarioBenchmark.");
+            teardownDelegate(configuration.Scenario);
 
+            if (configuration.SaveResults)
+            {
+                WriteResults(configuration.Scenario, fileNameWithoutExtension);
+            }
+        }
+
+        public void WriteResults(ScenarioBenchmark scenario, string fileNameWithoutExtension)
+        {
+            WriteXmlResults(scenario, fileNameWithoutExtension);
+
+            WriteTableResults(new[] { scenario }, fileNameWithoutExtension, false);
+        }
+
+        public void WriteXmlResults(ScenarioBenchmark scenario, string fileNameWithoutExtension)
+        {
             var xmlFileName = $"{fileNameWithoutExtension}.xml";
-            scenarioBenchmark.Serialize(xmlFileName);
+            scenario.Serialize(xmlFileName);
             OutputFileCallback?.Invoke(xmlFileName);
```

Comment:
Could you please share example xml files when you append to the Markdown table? There should still be one file per scenario.

Reply:
I'm still writing a separate XML file for each scenario, but I'm aggregating multiple scenarios for the Markdown and CSV files.  Here's the code: https://github.com/dsplaisted/sdk/blob/916ffbe41f910a2c729ef84f735e3ac3bbd5752a/test/Microsoft.NET.Perf.Tests/PerfTest.cs#L123-L133